daq_main_common.py

```diff
@@ -29,10 +31,64 @@
 immediate_command_list = []
 z = 25
 
+def setGovState(state):
+  setGovRobot(gov_robot, state)
+
+
+functions = [anneal,
+  set_beamsize,
+  importSpreadsheet,
+  mvaDescriptor,
+  setTrans,
+  loop_center_xrec,
+  autoRasterLoop,
+  snakeRaster,
+  ispybLib.insertRasterResult,
+  backlightBrighter,
+  backlightDimmer,
+  changeImageCenterHighMag,
+  changeImageCenterLowMag,
+  center_on_click,
+  runDCQueue,
+  warmupGripper,
+  dryGripper,
+  enableDewarTscreen,
+  parkGripper,
+  stopDCQueue,
+  continue_data_collection,
+  mountSample,
+  unmountSample,
+  reprocessRaster,
+  fastDPNodes,
+  spotNodes,
+  unmountCold,
+  openPort,
+  set_beamcenter,
+  closePorts,
+  clearMountedSample,
+  recoverRobot,
+  rebootEMBL,
+  restartEMBL,
+  openGripper,
+  closeGripper,
+  homePins,
+  setSlit1X,
+  setSlit1Y,
+  testRobot,
+  setGovState]
+
+whitelisted_functions: "Dict[str, Callable]" = {
+    func.__name__: func for func in functions
+}
   
 def execute_command(command_s):
-  logger.info('executing command: %s' % command_s)
-  exec(command_s)
+  logger.info("executing command: %s" % command_s)
+  try:
+      command: "dict[str, Any]" = json.loads(command_s)
+      func = whitelisted_functions[command["function"]]
+      func(*command["args"], **command["kwargs"])
+  except Exception as e:
+      logger.error(f"Error executing {command_s}: {e}")
```

Comment:
Would it be help to use `logger.exception` here, which shows both the log message and the full traceback? I suppose it depends on how user-facing these logs are.